### PR TITLE
Add a pipeline step to calculate precisions and recalls for PII recognition

### PR DESCRIPTION
**Pull request by gabechu**
### Description
This is another pipeline step that calculates precisions and recalls via comparing ground truth and predictions. Here's the changes
1) Added dataclasses as the containers to hold precision and recall scores. The alternative is to use plain list which could be a deep nested structure. Tests are updated to reflect the change of data structure.
2) Added `build_label_mapping` function. This function is responsible for creating entity label to integer mapping.
3) Added the pipeline step for calculating precessions and recalls.

#### Checklist
- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: pii_recognition/evaluation/character_level_evaluation.py [hunk 1]
```diff
@@ -9,43 +9,94 @@
...
+            continue
         s = span.start
         e = span.end
+
         if e > text_length:
             raise ValueError(
                 f"Entity span index is out of range: text length is "
                 f"{text_length} but got span index {e}."
             )
         label_name = span.entity_type
```
Comment (chausler) [inline pii_recognition/evaluation/character_level_evaluation.py:99]: this is duplicated

File: pii_recognition/evaluation/character_level_evaluation.py [hunk 2]
```diff
@@ -9,43 +9,94 @@
...
+@dataclass
+class EntityRecall:
+    entity: Entity
+    recall: float
+    entity_src: str = "ground_truth"
+
+
+@dataclass
+class TicketScore:
+    precisions: List[EntityPrecision]
```
Comment (chausler) [inline pii_recognition/evaluation/character_level_evaluation.py:28]: we should probably call this something my generic like `textscore` or `entityscore` ... i'm not sure of a good name, but the text doesn't have to be a Ticket :-D

Comment (gabechu) [reply pii_recognition/evaluation/character_level_evaluation.py:28]: Agreed. I like the name `TextScore`, will change to this name then.

File: pii_recognition/evaluation/character_level_evaluation.py (hunk 1 above)
Comment (gabechu) [reply pii_recognition/evaluation/character_level_evaluation.py:99]: Good catch! Removed.

**Review by chausler — APPROVED** [review]

**Pull request merged**